api/v1beta1/clusterclass_types.go

```diff
@@ -399,6 +400,7 @@ type PatchSelectorMatch struct {
 // in specific MachineDeploymentClasses in .spec.workers.machineDeployments.
 type PatchSelectorMatchMachineDeploymentClass struct {
 	// Names selects templates by class names.
+	// Note: If the list is empty then it will match all MachineDeploymentClass.
```

Comment:
With the new checks, this comment is no longer applicable. Can you please delete it?